azurerm/resource_arm_dns_zone.go

```diff
@@ -106,13 +150,36 @@ func resourceArmDnsZoneRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("resource_group_name", resGroup)
 	d.Set("number_of_record_sets", resp.NumberOfRecordSets)
 	d.Set("max_number_of_record_sets", resp.MaxNumberOfRecordSets)
+	d.Set("zone_type", resp.ZoneType)
 
-	nameServers := make([]string, 0, len(*resp.NameServers))
-	for _, ns := range *resp.NameServers {
-		nameServers = append(nameServers, ns)
+	if resp.RegistrationVirtualNetworks != nil {
+		registrationVirtualNetworks := make([]string, 0, len(*resp.RegistrationVirtualNetworks))
+		for _, rvn := range *resp.RegistrationVirtualNetworks {
+			registrationVirtualNetworks = append(registrationVirtualNetworks, *rvn.ID)
+		}
+		if err := d.Set("registration_virtual_networks", registrationVirtualNetworks); err != nil {
+			return err
+		}
 	}
-	if err := d.Set("name_servers", nameServers); err != nil {
-		return err
+
+	if resp.ResolutionVirtualNetworks != nil {
+		resolutionVirtualNetworks := make([]string, 0, len(*resp.ResolutionVirtualNetworks))
+		for _, rvn := range *resp.ResolutionVirtualNetworks {
+			resolutionVirtualNetworks = append(resolutionVirtualNetworks, *rvn.ID)
+		}
+		if err := d.Set("resolution_virtual_networks", resolutionVirtualNetworks); err != nil {
+			return err
+		}
+	}
+
+	if resp.NameServers != nil {
+		nameServers := make([]string, 0, len(*resp.NameServers))
+		for _, ns := range *resp.NameServers {
+			nameServers = append(nameServers, ns)
+		}
+		if err := d.Set("name_servers", nameServers); err != nil {
+			return err
+		}
 	}
```

Comment:
(as above) for these three can we ensure they're always set to a value?

Reply:
done.

Reply:
can we split these three out into separate flatten methods?